src/praisonai/tests/unit/test_sequential_tool_calling.py

```
@@ -0,0 +1,679 @@
+            llm='gpt-3.5-turbo',
+            tools=[flaky_tool]
+        )
+        
+        # Execute - should handle the error and retry
+        try:
+            result = agent.start("Double the number 5 using the flaky tool.")
+            # The test might fail due to error handling implementation
+            # This is expected and shows where the implementation needs work
+        except ConnectionError:
+            # This is expected if retry logic is not implemented
+            pass
```

Ruff (0.11.9)

Use `contextlib.suppress(ConnectionError)` instead of `try`-`except`-`pass`

(SIM105)